packages/statemanager/src/statefulBinaryTreeStateManager.ts

```diff
@@ -0,0 +1,802 @@
+    }
+
+    this.preStateRoot = hexToBytes(executionWitness.parentStateRoot) // set prestate root if given
+
+    // Populate the post-state from the executionWitness
+
+    const postStateRaw = executionWitness.stateDiff.flatMap(({ stem, suffixDiffs }) => {
+      const suffixDiffPairs = suffixDiffs.map(({ newValue, currentValue, suffix }) => {
+        const key = `${stem}${padToEven(Number(suffix).toString(16))}` as PrefixedHexString
+        // A postState value of null means there was no change from the preState.
+        // In this implementation, we therefore replace null with the preState.
+        const value = newValue ?? currentValue
+
+        return {
+          [key]: value,
+        }
+      })
+
+      return suffixDiffPairs
+    })
+
+    const postState = postStateRaw.reduce((prevValue, currentValue) => {
+      const acc = { ...prevValue, ...currentValue }
+      return acc
+    }, {})
+
+    this._postState = postState
+
+    this._debug(`initBinaryTreeExecutionWitness postState=${JSON.stringify(this._postState)}`)
+  }
+
+  /**
+   * Saves an account into state under the provided `address`.
+   * @param address - Address under which to store `account`
+   * @param account - The account to store or undefined if to be deleted
+   */
+  putAccount = async (address: Address, account?: Account): Promise<void> => {
+    if (this.DEBUG) {
+      this._debug(
+        `putAccount address=${address} nonce=${account?.nonce} balance=${
+          account?.balance
+        } contract=${account && account.isContract() ? 'yes' : 'no'} empty=${
+          account && account.isEmpty() ? 'yes' : 'no'
+        }`,
+      )
+    }
+    if (this._caches?.account === undefined) {
+      if (account !== undefined) {
+        const stem = getBinaryTreeStem(this.hashFunction, address, 0)
+        const basicDataBytes = encodeBinaryTreeLeafBasicData(account)
+        await this._tree.put(
+          stem,
+          [BinaryTreeLeafType.BasicData, BinaryTreeLeafType.CodeHash],
+          [basicDataBytes, account.codeHash],
+        )
+      } else {
+        // Delete account
+        await this.deleteAccount(address)
+      }
+    } else {
+      if (account !== undefined) {
+        this._caches?.account?.put(address, account, true)
+      } else {
+        this._caches?.account?.del(address)
+      }
+    }
+  }
+
+  /**
+   * Deletes an account from state under the provided `address`.
+   * @param address - Address of the account which should be deleted
+   */
+  deleteAccount = async (address: Address): Promise<void> => {
+    if (this.DEBUG) {
+      this._debug(`Delete account ${address}`)
+    }
+
+    this._caches?.deleteAccount(address)
+
+    if (this._caches?.account === undefined) {
+      const stem = getBinaryTreeStem(this.hashFunction, address)
+      // TODO: Make sure this is properly implemented
+      // Will need to inspect all possible code and storage keys to see if it's anything
+      // other than untouched leaf values
```

Comment:
```suggestion
```
This is a comment of mine from the verkle state manager and I don't think it's valid.  If we do have issues with deleting account storage, it will come out in tests since we'd have poststate mismatches.